pennylane/transforms/qmc.py

```diff
@@ -0,0 +1,353 @@
+    defined in `this <https://arxiv.org/abs/1805.00109>`__ paper.
+
+    Given a callable ``fn`` input corresponding to the :math:`\mathcal{F}` unitary in the above
+    paper, this function transforms the circuit into a controlled-version of the :math:`\mathcal{Q}`
+    unitary which forms part of the quantum Monte Carlo algorithm. In this algorithm, one of the
+    wires acted upon by :math:`\mathcal{F}`, specified by ``target_wire``, is used to embed a
+    Monte Carlo estimation problem. The :math:`\mathcal{Q}` unitary is then designed to encode the
+    target expectation value as a phase in one of its eigenvalues.
+
+    This function transforms to a controlled version of :math:`\mathcal{Q}` that is compatible with
+    quantum phase estimation (see :class:`~.QuantumPhaseEstimation` for more details).
+
+    Args:
+        fn (Callable): a quantum function that applies quantum operations according to the
+            :math:`\mathcal{F}` unitary used as part of quantum Monte Carlo estimation
+        wires (Union[Wires or Sequence[int]]): the wires acted upon by the ``fn`` circuit
+        target_wire (Union[Wires, int]): The wire in which the expectation value is encoded. Must be
+            contained within ``wires``.
+        control_wire (Union[Wires, int]): the control wire from the register of phase estimation
+            qubits
+        work_wires (Union[Wires, Sequence[int], or int]): additional work wires used when
+            decomposing :math:`\mathcal{Q}`
+
+    Returns:
+        function: The input function transformed to the :math:`\mathcal{Q}` unitary
+
+    Raises:
+        ValueError: if ``target_wire`` is not in ``wires``
+    """
+    fn_inv = adjoint(fn)
+
+    wires = Wires(wires)
+    target_wire = Wires(target_wire)
+    control_wire = Wires(control_wire)
+    work_wires = Wires(work_wires)
+
+    if not wires.contains_wires(target_wire):
+        raise ValueError("The target wire must be contained within wires")
+
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+        _apply_controlled_v(target_wire=target_wire, control_wire=control_wire)
+        fn_inv(*args, **kwargs)
+        _apply_controlled_z(wires=wires, control_wire=control_wire, work_wires=work_wires)
+        fn(*args, **kwargs)
+
+        _apply_controlled_v(target_wire=target_wire, control_wire=control_wire)
+        fn_inv(*args, **kwargs)
+        _apply_controlled_z(wires=wires, control_wire=control_wire, work_wires=work_wires)
+        fn(*args, **kwargs)
+
+    return wrapper
+
+
+def quantum_monte_carlo(fn, wires, target_wire, estimation_wires):
+    r"""Provides the circuit to perform the
+    `quantum Monte Carlo estimation <https://arxiv.org/abs/1805.00109>`__ algorithm.
+
+    The input ``fn`` should be the quantum circuit corresponding to the :math:`\mathcal{F}` unitary
+    in the paper above that encodes the probability distribution and random variable onto ``wires``
```

Comment:
Consider splitting this long sentence into two

Reply:
Thanks, have split.